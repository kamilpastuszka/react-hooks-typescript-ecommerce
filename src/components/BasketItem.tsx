import React from "react";
import { FaTrashAlt } from "react-icons/fa";

export default function BasketItem(item: {
  name: string;
  price: number;
  removeProduct: any;
}) {
  return (
    <tr>
      <td>{item.name}</td>
      <td>{item.price}</td>
      <td>
        <span onClick={item.removeProduct}>
          <FaTrashAlt />
        </span>
      </td>
    </tr>
  );
}
