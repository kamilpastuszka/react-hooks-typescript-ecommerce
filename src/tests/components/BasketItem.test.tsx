import React from "react";
import { shallow } from "enzyme";
import BasketItem from "../../components/BasketItem";
import toJson from "enzyme-to-json";

test("It should render BasketItem component correctly", () => {
  const onRemoveProductSpy = jest.fn();
  const wrapper = shallow(
    <BasketItem
      name={"Item 1"}
      price={100}
      removeProduct={onRemoveProductSpy}
    />
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
