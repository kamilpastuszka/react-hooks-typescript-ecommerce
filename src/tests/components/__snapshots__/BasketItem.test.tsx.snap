// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Basket correctly 1`] = `
<tr>
  <td>
    Item 1
  </td>
  <td>
    100
  </td>
  <td>
    <span
      onClick={[MockFunction]}
    >
      <FaTrashAlt />
    </span>
  </td>
</tr>
`;
